junebug/api.py

```diff
@@ -25,15 +41,46 @@ class ApiUsageError(JunebugError):
     code = http.BAD_REQUEST
 
 
+class RabbitmqManagementClient(object):
+
+    def __init__(self,  base_url, username, password, http_client):
+        self.base_url = base_url
+        self.username = username
+        self.password = password
+        self.http_client = http_client
+
+    def get_queue(self, vhost, queue_name):
+        url = 'http://%s/api/queues/%s/%s' % (
+            self.base_url,
+            urllib.quote(vhost, safe=''),
+            queue_name
+        )
+
+        d = self.http_client.get(url, auth=(self.username, self.password))
+        d.addCallback(treq.json_content)
+        return d
+
+
 class JunebugApi(object):
     app = Klein()
 
+    clock = reactor
+
     def __init__(self, service, config):
         self.service = service
         self.redis_config = config.redis
         self.amqp_config = config.amqp
         self.config = config
 
+    @classmethod
+    def pool_factory(cls, reactor):
+        pool = HTTPConnectionPool(reactor, persistent=True)
+        pool.maxPersistentPerHost = TPS_LIMIT
+
+    @classmethod
+    def agent_factory(cls, reactor, pool=None):
+        return Agent(reactor, pool=pool)
```

Comment:
I think that the client can have its own pool and agent, and manage its own connection pool.